packages/sign-client/src/controllers/engine.ts

```diff
@@ -572,6 +591,402 @@ export class Engine extends IEngine {
     return this.client.pendingRequest.getAll();
   };
 
+  // ---------- Auth ------------------------------------------------ //
+
+  public authenticate: IEngine["authenticate"] = async (params) => {
+    this.isInitialized();
+    this.isValidAuthenticate(params);
+
+    const { chains, statement = "", uri, domain, nonce, type, exp, nbf, methods = [] } = params;
+    // reassign resources to remove reference as the array is modified and might cause side effects
+    const resources = [...(params.resources || [])];
```

Comment:
This can be simplified and avoid spreading over the array.
```suggestion
    const resources = params.resources ?? [];
```

Reply:
It was done as you suggest but as described in the comment, we modify the `resources` array which might not be expected by the implementing party leading to side effects